### PR TITLE
feat: enable comment incentives on closed issues

### LINKED ISSUE #754: Enable Comment Incentives On "Closed As Complete" Parent Issues
When creating complex projects, usually it is most convenient to start drafting an issue and then later break it down into child issues. 

As we know, we can not have a bounty hunter directly work on a parent issue (we disable permit generation on parent issues) 

However it is redundant for the author to copy and rewrite the context in the child issue bodies. 

We should enable the permit generation for the comment incentives on the parent issues. 

This means 1. issue creator incentives and 2. comment incentives.

### PR DESCRIPTION
**Pull request by EtherealGlow**
Resolves #754

Quality Assurance: None for now

### CODE REVIEW

**Comment by EtherealGlow** [comment]
i cant see anything that would stop a permit from generating in post
@pavlovcik

**Review by cuddle-dev — CHANGES_REQUESTED** [review]
Looks like its still in progress. converted it to a draft.

**Comment by EtherealGlow** [comment]
@0xcodercrane i cant see anything that would make a incentive permit to stop generating

**Comment by 0xcodercrane** [comment]
> @0xcodercrane i cant see anything that would make a incentive permit to stop generating

https://github.com/ubiquity/ubiquibot/blob/development/src/handlers/pricing/action.ts#L17

can you see this line? @EtherealGlow

**Comment by EtherealGlow** [comment]
But what about incentives tho, incentives don't have a price label

**Comment by wannacfuture** [comment]
> @0xcodercrane But what about incentives tho, incentives don't have a price label

you can refer `incentivizeComments` and `incentivizeCreatorComment`

**Comment by EtherealGlow** [comment]
https://github.com/EtherealGlow/ubiquibot/issues/18 on this it tried generating conversation rewards.
https://github.com/EtherealGlow/ubiquibot/issues/16 on this it generated conversation rewards after reopening the issue

**Comment by EtherealGlow** [comment]
@wannacfuture can you review this

**Comment by pavlovcik** [comment]
> https://github.com/EtherealGlow/ubiquibot/issues/18 on this it tried generating conversation rewards.
> 
> https://github.com/EtherealGlow/ubiquibot/issues/16 on this it generated conversation rewards after reopening the issue 

This doesn't generate rewards on the meta issue why are you submitting this as completed work?

**Comment by EtherealGlow** [comment]
Meta issue?

**Comment by EtherealGlow** [comment]
man now somehow my permits are broken, @Draeieg can you help me qa this because i cannot

**Comment by EtherealGlow** [comment]
> > [EtherealGlow#18](https://github.com/EtherealGlow/ubiquibot/issues/18) on this it tried generating conversation rewards.
> > [EtherealGlow#16](https://github.com/EtherealGlow/ubiquibot/issues/16) on this it generated conversation rewards after reopening the issue
> 
> This doesn't generate rewards on the meta issue why are you submitting this as completed work?

it was prob because all incentive elements were set to 0
my fault on that one

**Comment by whilefoo** [comment]
@EtherealGlow what's the status of this PR? If it's done, can you post QA? Also resolve the conflicts

**Comment by EtherealGlow** [comment]
Incentives aren't working even if I do getPrivateKey(cipher) directly

**Review by whilefoo — COMMENTED** [review]
I think we shouldn't generate permit again if the issue is reopened and closed because we have one permit per user per issue rule so the user will only be able to claim one permit. Also I don't see this being part of the specification - it's only about generating issue creator and conversation rewards for parent issues.

@EtherealGlow try pulling latest changes from development because there was a big change to incentives

File: src/handlers/payout/post.ts [hunk 1]
```diff
@@ -42,10 +41,6 @@ export const calculateIssueConversationReward = async (calculateIncentives: Ince
 
   const assignees = issue?.assignees ?? [];
   const assignee = assignees.length > 0 ? assignees[0] : undefined;
```
Comment (whilefoo) [inline src/handlers/payout/post.ts:35]: `assignees` and `assignee` is not used anymore if I see correctly

**Review by whilefoo — CHANGES_REQUESTED** [review]
Can you post QA? 
Also please learn how to use Git without force pushing, we won't be able to process your future PRs if you keep using force push

**Comment by EtherealGlow** [comment]
[qa](https://github.com/EtherealGlow/ubiquibot/pull/25)

**Comment by EtherealGlow** [comment]
the bot requested review from me but idk why the bot requested 2 times, does someone know why?

**Comment by EtherealGlow** [comment]
will do this shortly,
fixing the bot is harder than any issue i have ever done

**Comment by EtherealGlow** [comment]
@whilefoo how would i qa, my wallet doesnt have enough balance and it refuses to generate a permit

**Comment by wannacfuture** [comment]
> @whilefoo how would i qa, my wallet doesnt have enough balance and it refuses to generate a permit

@EtherealGlow Bot is able to generate permit URL even you don't have enough balance.

You cannot just claim it when you go to that url(saying insuficcient fund).

**Comment by EtherealGlow** [comment]
> > @whilefoo how would i qa, my wallet doesnt have enough balance and it refuses to generate a permit
> 
> @EtherealGlow Bot is able to generate permit URL even you don't have enough balance.
> 
> You cannot just claim it when you go to that url(saying insuficcient fund).

https://github.com/me505/ubiquibot/issues/10#issuecomment-1736926872

**Comment by wannacfuture** [comment]
> > > @whilefoo how would i qa, my wallet doesnt have enough balance and it refuses to generate a permit
> > 
> > 
> > @EtherealGlow Bot is able to generate permit URL even you don't have enough balance.
> > You cannot just claim it when you go to that url(saying insuficcient fund).
> 
> [me505#10 (comment)](https://github.com/me505/ubiquibot/issues/10#issuecomment-1736926872)

Please take a closer look at the `generatePermitForComments` function.  @me505 
I don't think its releated to the funds in your wallet. There is no feature to check the funds in the wallet in that function.

The reason of that error message is not your insufficient funds in your wallet.
I bet it is related to missing account or the reward value is 0 or higher than maxPermitPrice.

**Comment by EtherealGlow** [comment]
@wannacfuture any idea why the result of generatePermitForComments is undefined?

**Comment by wannacfuture** [comment]
> @wannacfuture any idea why the result of generatePermitForComments is undefined?

Like I said, I'm assuming that the reward value is calculated as 0 or account or missing or is higher than the maxPermitPrice.

You can check it adding console logs to that function.

**Pull request closed without merging**